### ISSUE TITLE
Add page links for APIs mentioned in the docs

### ISSUE THREAD

**Issue by jon-oracle** [openssl/openssl#12392]
The manual pages have a lot of references to functions which are not links, despite the function being documented in another page.

An example would be in the SSL_SESSION_free page where SSL_get1_session() is mentioned. Adding a link of the form L<SSL_get1_session()|SSL_get_session(3)> links to the page containing that API. Note that the API & link name may or may not be the same.

It would be really useful to be able to jump between function docs without having to search the index. I have a script which can do the conversion in a (mostly) automated way. Results for SSL APIs can be viewed here:

https://github.com/openssl/openssl/commit/128f716076d4861b826da9fe131d5a44ce1f24d1

**Comment by levitte** [comment]
`L<SSL_get_session(3)>` would be more in line with how man-pages are usually written.  man-pages(7) isn't absolutely clear on it, rather takes that for granted, but you do see a hint in this section:

> **Formatting conventions for manual pages describing functions**
> ...
>       Any reference to the subject of the current manual page should be
>       written with the name in bold followed by a pair of parentheses in
>       Roman (normal) font.  For example, in the **fcntl**(2) man page,
>       references to the subject of the page would be written as: **fcntl**().

**Comment by jon-oracle** [comment]
> `L<SSL_get_session(3)>` would be more in line with how man-pages are usually written. man-pages(7) isn't absolutely clear on it, rather takes that for granted, but you do see a hint in this section:
> 
> > **Formatting conventions for manual pages describing functions**
> > ...
> > Any reference to the subject of the current manual page should be
> > written with the name in bold followed by a pair of parentheses in
> > Roman (normal) font.  For example, in the **fcntl**(2) man page,
> > references to the subject of the page would be written as: **fcntl**().

L<SSL_get_session(3)> works if the page name and API name are the same, but a majority of pages document a collection of APIs, not just the one the page is named for. Using the L<api_name|link_page(3)>() format is the best I've come up with so far, I'm open to suggestions though.

**Comment by richsalz** [comment]
Many man3 manpages document multiple API's.  All of those API's should be listed in the NAME and SYNOPSIS sections else find-doc-nits will complain.  Installing the manpage makes links in the install directory.  So *every* API foo(3) [or foo(7)] even if it is part of doc/man3/bar.pod, will exist.

TL;DR, using ```L<foo(3)>``` should always work.  If not, its a bug.

**Comment by levitte** [comment]
> `L<SSL_get_session(3)>` works if the page name and API name are the same, but a majority of pages document a collection of APIs...

... and?  As long as it's named in any of the pod files (in the NAMES section), the link should work.  Incidently, I just grepped for `SSL_get_session`, and it turns out that we already have `L<SSL_get_session(3)>` form links...

**Comment by jon-oracle** [comment]
> Many man3 manpages document multiple API's. All of those API's should be listed in the NAME and SYNOPSIS sections else find-doc-nits will complain. Installing the manpage makes links in the install directory. So _every_ API foo(3) [or foo(7)] even if it is part of doc/man3/bar.pod, will exist.
> 
> TL;DR, using `L<foo(3)>` should always work. If not, its a bug.

Ahh ok, that makes sense. In that case yes the links can be just L<[func](3)>. I'll try generating this for each function instead.

**Comment by nhorman** [comment]
Marking as inactive to be closed at the end of 3.4 dev, barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.
